amip_interp/include/amip_interp.inc

```diff
@@ -452,25 +114,25 @@ if ( .not.use_daily ) then
               Interp % data1 = Interp % data2
               temp1(:,:) = temp2(:,:)   ! SJL BUG fix: June 24, 2011
           else
-              call read_record ('sst', Date1, Udate1, temp1)
+              call READ_RECORD_ ('sst', Date1, Udate1, temp1)
               if ( use_ncep_sst .and. (.not. no_anom_sst) ) then
-                   temp1(:,:) = temp1(:,:) + sst_anom(:,:)
+                   temp1(:,:) = temp1(:,:) + SST_ANOM_ (:,:)
               endif
               call horiz_interp ( Interp%Hintrp, temp1, Interp%data1 )
-              call clip_data ('sst', Interp%data1)
+              call CLIP_DATA_ ('sst', Interp%data1)
              Interp % Date1 = Date1
           endif
       endif
 
 !-----------------------------------------------------------------------
 
       if (Date2 /= Interp % Date2) then
-          call read_record ('sst', Date2, Udate2, temp2)
+          call READ_RECORD_ ('sst', Date2, Udate2, temp2)
           if ( use_ncep_sst .and. (.not. no_anom_sst) ) then
-               temp2(:,:) = temp2(:,:) + sst_anom(:,:)
+               temp2(:,:) = temp2(:,:) + SST_ANOM_ (:,:)
```

Comment:
For the case where sst_anom_r4 is used... that means sst_anom_r8 is not used and the pointer sst_anom is pointing to sst_anom_r8, how does the user retrieve the sst_anom_r4 data?

Reply:
I think it's not possible to allow both `r4_kind` and `r8_kind` access in a way that's backward-compatible. We could make `sst_anom` and `sst_ncep` both `class(*), allocatable`, but then any user code that accesses them would have to be wrapped in a `select type` block.